### ISSUE TITLE
`trailing-comma` triggered on chained method call

### ISSUE THREAD

**Issue by mrhania** [palantir/tslint#1852]
### Bug Report

- __TSLint version__: `4.0.2`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
function quux(f: any) {
}

quux((foo: any) => foo
    .bar()
    .baz() // missing trailing comma here
);
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "trailing-comma": [true, {
      "multiline": "always",
      "singleline": "never"
    }]
  }
}
```

#### Actual behavior

Linting ends with an error:

    foo.ts[6, 10]: Missing trailing comma

Rewriting the code like this:

```typescript
quux((foo: any) => foo
    .bar()
    .baz(),
);
```

fixes the problem (but looks pretty weird).

#### Expected behavior

No linting error should be reported.

**Comment by adidahiya** [comment]
This is expected behavior. Trailing commas are required in multiline argument lists. Agreed that this is a slightly weird edge case, but I'm not sure it's worth bending the rule implementation to change the behavior here.

**Comment by mrhania** [comment]
Sounds reasonable. Now that you explained the behaviour, my example can be simplified to this:

``` typescript
let foo = {} as any;
quux(foo
    .bar()
    .baz()
);
```

So the rule could be something like _"commas are required in multiline argument lists except for single-item lists with first token on the same line as method call"_. On the other hand, that sounds overly complicated and perhaps unnecessary. The initial version of my example could be rewritten like so:

```typescript
quux((foo: any) => {
    return foo
        .bar()
        .baz();
});
```

Actually, I might even prefer this version over the first one. If the lambda body doesn't fit a single line, perhaps it shouldn't be a non-braced lambda at all (an idea for a new rule?). Because of that, I think it is reasonable to close this issue.
